sql/functions/run_maintenance.sql

```diff
@@ -219,11 +221,21 @@ LOOP
         FOR v_row_max_time IN
             SELECT partition_schemaname, partition_tablename FROM @extschema@.show_partitions(v_row.parent_table, 'DESC', false)
         LOOP
-            EXECUTE format('SELECT %s::text FROM %I.%I LIMIT 1'
-                                , v_partition_expression
-                                , v_row_max_time.partition_schemaname
-                                , v_row_max_time.partition_tablename
-                            ) INTO v_child_timestamp;
+
+            IF v_control_type = 'time' OR (v_control_type = 'id' AND v_row.epoch <> 'none') THEN
+                EXECUTE format('SELECT %s::text FROM %I.%I'
```

Comment:
Missing the LIMIT 1 here

Reply:
Ah, can't believe I missed that. Have fixed it now.
Maintenance was still working though, select into seems to be taking the first row in any case